core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -582,6 +592,53 @@ private void maybeTerminateChannel() {
     }
   }
 
+  @Override
+  public ConnectivityState getState(boolean requestConnection) {
+    ConnectivityState savedChannelState = channelState;
+    if (requestConnection && savedChannelState == IDLE) {
+      channelExecutor.executeLater(new Runnable() {
+        @Override
+        public void run() {
+          exitIdleMode();
+        }
+      }).drain();
+    }
+    return savedChannelState;
+  }
+
+  private static final class StateChangeCallBack implements Runnable {
+    final ConnectivityState source;
+    final Runnable callback;
+    boolean done;
```

Comment:
It's class is private and final, so any access specifier before `boolean done` effectively implies the same access scope.